sktime/forecasting/model_selection/tests/test_tune.py

```diff
@@ -9,6 +9,7 @@
 from typing import Union
 
 import numpy as np
+import optuna
```

Comment:
Please do not add import of a soft dependency here. It'll fail CI that tests whether the package can be imported and used without any soft dependencies. 

You can import inside a function or method protected by checking of soft dependencies. You'll find similar examples throughout the codebase. Let me know if you need help.

Reply:
thanks for the feedback, just for my understanding, are you talking about something like the code below? Otherwise could you give me a hint where to look in the codebase?
My issue is at parametrize sampler step I need to have optuna imported

`
from sktime.utils.dependencies import _check_soft_dependencies

def load_optuna_dependency():
    if _check_soft_dependencies("optuna"):
        import optuna

        return optuna

optuna = load_optuna_dependency()
`

Reply:
hm, a solution would be to use `pytest` fixtures then, and import optuna only inside the fixture.

Reply:
Not 100% sure, that is what you expected, but it works that way if importing in a fixture